lib/streamlit/elements/camera_input.py

```diff
@@ -0,0 +1,198 @@
+# Copyright 2018-2021 Streamlit Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from streamlit.type_util import Key, to_key
+from textwrap import dedent
+from typing import Optional, cast, List
+
+import streamlit
+from streamlit.proto.CameraInput_pb2 import (
+    CameraInput as CameraInputProto,
+)
+
+from streamlit.report_thread import get_report_ctx
+from streamlit.state.widgets import register_widget
+from streamlit.state.session_state import (
+    WidgetArgs,
+    WidgetCallback,
+    WidgetKwargs,
+)
+
+from ..proto.Common_pb2 import (
+    FileUploaderState as FileUploaderStateProto,
+    UploadedFileInfo as UploadedFileInfoProto,
+)
+from ..uploaded_file_manager import UploadedFile, UploadedFileRec
+
+from .form import current_form_id
+from .utils import check_callback_rules, check_session_state_rules
+
+SomeUploadedSnapshotFile = Optional[UploadedFile]
+
+
+class CameraInputMixin:
+    def camera_input(
+        self,
+        label: str,
+        key: Optional[Key] = None,
+        help: Optional[str] = None,
+        on_change: Optional[WidgetCallback] = None,
+        args: Optional[WidgetArgs] = None,
+        kwargs: Optional[WidgetKwargs] = None,
+        *,  # keyword-only arguments:
+        disabled: bool = False,
+    ) -> SomeUploadedSnapshotFile:
+        """Display a widget that returns pictures from the user's webcam.
+        Parameters
+        ----------
+        label : str
+            A short label explaining to the user what this camera input is for.
+
+        key : str or int
+            An optional string or integer to use as the unique key for the widget.
+            If this is omitted, a key will be generated for the widget
+            based on its content. Multiple widgets of the same type may
+            not share the same key.
+
+        help : str
+            A tooltip that gets displayed next to the file uploader.
```

Comment:
nit: most other widget docstrings mention that this parameter is optional. It seems like `file_uploader` doesn't, but I think that one should probably be changed to be consistent with other widgets as well.